hathor/indexes/base_index.py

```diff
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from __future__ import annotations
```

Comment:
Why is it needed?

Reply:
To import `HathorSettings` only for type checking. Before, we would do `"HathorSettings"`, but I'm now pushing into using this which allows using the type name normally, without quotes.

Reply:
Got it. Thanks!